src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,450 @@ title: Astro Loader API
+      store.clear();
+
+      for (const item of feed.items) {
+        const data = await parseData({
+          id: item.guid,
+          data: item,
+        });
+
+        const digest = generateDigest(data);
+
+        store.set({
+          id,
+          data,
+          digest,
+        });
+      }
+    },
+  };
+}
+```
+
+#### `watcher`
+
+<p>
+
+**Type**: `FSWatcher`
+</p>
+
+When running in dev mode, this is a filesystem watcher that can be used to trigger updates. See [`ViteDevServer`](https://vitejs.dev/guide/api-javascript.html#vitedevserver) for more information.
+
+```ts title="Extract from the file() loader" {8-13}
+return {
+  name: 'file-loader',
+  load: async ({ config, store, watcher }) => {
+    const url = new URL(fileName, config.root);
+    const filePath = fileURLToPath(url);
+    await syncData(filePath, store);
+
+    watcher?.on('change', async (changedPath) => {
+      if (changedPath === filePath) {
+        logger.info(`Reloading data from ${fileName}`);
+        await syncData(filePath, store);
+      }
+    });
+  },
+};
+```
+
+#### `refreshContextData`
+
+<p>
+
+**Type**: `Record<string, unknown>`
+</p>
+
+If the loader has been triggered by an integration, this may optionally contain extra data set by that integration. It is only set when the loader is triggered by an integration. See the [`astro:server:setup`](/en/reference/integrations-reference/#refreshcontent-option) hook reference for more information.
+
+```ts title=loader.ts {5-8}
+export function myLoader(options: { url: string }): Loader {
+  return {
+    name: "my-loader",
+    load: async ({ refreshContextData, store, logger }) => {
+      if(refreshContextData?.webhookBody) {
+        logger.info("Webhook triggered with body");
+        processWebhook(store, refreshContextData.webhookBody);
+      }
+      // ...
+    },
+  };
+}
+```
+
+### `DataStore`
+
+The data store is a loader's interface to the content collection data. It is a key value store, scoped to the collection, so a loader can only access the data for its own collection.
+
+#### `get`
+
+<p>
+
+**Type**: <code>(key: string) => <a href="#dataentry">DataEntry</a> | undefined</code>
+</p>
+
+Get an entry from the store by its ID. Returns `undefined` if the entry does not exist. 
+
+```ts
+const existingEntry = store.get("my-entry");
+```
+
+The returned object is a [`DataEntry`](#dataentry) object.
+
+#### `set`
+
+<p>
+
+**Type**: <code>(entry: <a href="#dataentry">DataEntry</a>) => boolean</code>
+</p>
+
+Adds an entry to the store, returning `true` if the entry was set. If `digest` was passed and matched an existing entry with the same ID, it will return `false` and not update the entry. See [`DataEntry`](#dataentry) for more information on the fields to pass. The entry should have been parsed using `parseData` before being stored. As well as applying any validations and transforms in the schema, this will also ensure that any images are correctly processed.
```

Comment:
```suggestion
Used after data has been [validated and parsed](#parseData) to add an entry to the store, returning `true` if the entry was set. This returns `false` when the [`digest`](#digest) property determines that an entry has not changed and should not be updated.
```
First paragraph: I feel like that was maybe a lot here? Does this adequately convey what's needed for *this* property with the link? Because we do mention that the `digest` property will search for matching entries, and I think that's adequately covered there?

Second paragraph: I always get nervous when I see "backwards documentation" :sweat_smile: (Oh, and you should have done this first....) I think this is important enough to put it right in the description!